### PR TITLE
fix: Always close main update PR if not merged

### PR DESCRIPTION
**Pull request by mikefat**
After the integration test resilience [PR](https://github.com/grafana/sigma-rule-deployment/pull/364) we're still seeing some edge cases where the state is left ambiguous. This PR aims to bring the integration tests to always be consistent - the state should always be the same when the integration tests pass or don't pass, without broken stages in between.

## What is the problem?

The token minted for the integration tests is only good for 30m by design, which is good. If the integration test update main PR is not merged within that window:
- The main update PR is left open forever
- Then, if we run it again the checkout-or-create-branch.sh force resets the PR branch to main momentarily. There is a race condition where if GH sees a PR where the branch's head is the same as main's, it auto closes the PR. We added the check [gh pr list --json id,title -H "$MAIN_UPDATE_BRANCH"](https://github.com/grafana/sigma-rule-deployment/blob/main/.github/workflows/build-docker.yml#L264) which should have caught this but it only sees open PRs. The workflow thinks the PR exists and it's open, then GH closes it, and then posts a commit to an already [closed pr](https://github.com/grafana/sigma-rule-deployment-integration-test/pull/794)

## How this fixes it
We update the action to programatically close the main update PR if not merged in time. When the action runs again, if there's an open PR, it will also close it, so that the state is consistent between runs. As a result, a new PR is always created when running the integration tests, and main update PRs are not left lingering.

- add a if: always() step that closes the main update PR after 25m
- Additionally, when running the integration tests we always check if there's an open main update PR from a previous run and we close it. Then we always create a new PR and each run should be independent from the previous one

### CODE REVIEW

File: .github/workflows/build-docker.yml [hunk 1]
```diff
@@ -261,58 +274,64 @@ jobs:
...
+          # number below keeps every `gh pr` call on this PR; a branch name also resolves to the
+          # closed PRs earlier runs left on the same branch.
+          MAIN_UPDATE_PR_URL=$(gh pr create \
+            --title "Integration Test Main Update ${GITHUB_PR_NUMBER}" \
+            --body "This PR updates the main branch to the relevant hash for grafana/sigma-rule-deployment#${GITHUB_PR_NUMBER}" \
+            --head "$MAIN_UPDATE_BRANCH" \
+            --base main \
+            --label "automated" \
+            --draft)
+          MAIN_UPDATE_PR="${MAIN_UPDATE_PR_URL##*/}"
```
Comment (FlomoN) [inline .github/workflows/build-docker.yml:287]: I didn't know bash could do that! Gotta keep that in mind :)

**Review by FlomoN — APPROVED** [review]
LGTM!
